app/model/scoreboard.js

```diff
@@ -0,0 +1,135 @@
+import Target from './target.js';
+import Round from './round.js';
+
+
+/**
+ * Keeps track of the scoreboard and it's affiliated stats
+ */
+export default class Scoreboard {
+
+    constructor() {
+        let targetTypes = ['20', '18', '19', '17', '16', '15', 'B'];
+        this._targets = [];
+        for(let targetType of targetTypes) {
+            this._targets.push({type: targetType, counter: new Target()});
+        }
+        this._activeTargetIndex = 0;
+        this._activeTarget = this._targets[this._activeTargetIndex];
+
+        this._round = new Round();
+    }
+
+    /**
+     * Marks a dart that hits. Moves to next target as appropriate.
+     * Ignored if scoreboard is already done.
+     * @returns {Scoreboard}
+     */
+    hit() {
+        if(!this.isDone()) {
+            this._round.nextDart();
+            this._activeTarget.counter.markHit({round: this._round.toString()});
+            if (this._activeTarget.counter.isDone()) {
+                this._activeTargetIndex++;
+                if (this._activeTargetIndex >= this._targets.length) {
+                    this._activeTarget = null;
+                }
+                else {
+                    this._activeTarget = this._targets[this._activeTargetIndex];
+                }
+            }
+        }
+        return this;
+    }
+
+    /**
+     * Marks a dart that hits.
+     * Ignored if scoreboard is already done
+     * @returns {Scoreboard}
+     */
+    miss() {
+        if(!this.isDone()) {
+            this._round.nextDart();
+        }
+        return this;
+    }
+
+    /**
+     * Finishes off the current round with misses; anywhere from 1 to 3 misses is registered.
+     * Ignored if scoreboard is already done.
+     * @returns {Scoreboard}
+     */
+    missx3() {
```

Comment:
This method name is quite specific to the round's THROWS_PER_ROUND const? Maybe "MissAll" or "MissRemainder" or something?